### PR TITLE
UCP: Add generic proxy_ep and use it as base for wireup_ep.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/4216/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@evgeny-leksikov @alex-mikheev can u pls review?

File: src/ucp/core/ucp_proxy_ep.c [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            proxy_ep->uct_ep = NULL;
+            break;
+        }
+    }
+
+    uct_ep_destroy(&proxy_ep->super);
+}
+
+void ucp_proxy_ep_set_uct_ep(ucp_proxy_ep_t *proxy_ep, uct_ep_h uct_ep,
+                             int is_owner)
```
Comment (evgeny-leksikov) [inline src/ucp/core/ucp_proxy_ep.c]: when is_owner may be == 0?

Comment (yosefe) [inline src/ucp/core/ucp_proxy_ep.c]: in next pull request which would introduce proxy_ep for sending signaled active message

**Pull request merged**